distrobuilder/main.go

```diff
@@ -587,32 +566,80 @@ EOF
 
 	mkdir -p /run/systemd/system/default.target.wants
 	ln -s /run/systemd/system/network-device-down.service /run/systemd/system/default.target.wants/network-device-down.service
-)
+}
 
-# Fix for systemd-networkd
-(
-	. /etc/os-release
+# fix_systemd_override_global generates the global systemd override
+fix_systemd_override_global() {
+	# Setup global override
+	dropin_dir="/run/systemd/system/service.d"
+	mkdir -p "${dropin_dir}"
+	echo "[Service]" > "${dropin_dir}/lxc.conf"
+	[ "${systemd_version}" -ge 247 ] && echo "ProtectProc=default" >> "${dropin_dir}/lxc.conf"
+	echo "ProtectControlGroups=no" >> "${dropin_dir}/lxc.conf"
+	echo "ProtectKernelTunables=no" >> "${dropin_dir}/lxc.conf"
+}
 
-	[ "${ID}" = "altlinux" ] || exit
+# fix_systemd_override_unit generates a unit specific override
+fix_systemd_override_unit() {
+	dropin_dir="/run/systemd/${1}.d"
+	mkdir -p "${dropin_dir}"
+	echo "[Service]" > "${dropin_dir}/lxc-service.conf"
+	[ "${systemd_version}" -ge 247 ] && echo "ProtectProc=default" >> "${dropin_dir}/lxc-service.conf"
+	[ "${systemd_version}" -ge 232 ] && echo "ProtectControlGroups=no" >> "${dropin_dir}/lxc-service.conf"
+	[ "${systemd_version}" -ge 232 ] && echo "ProtectKernelTunables=no" >> "${dropin_dir}/lxc-service.conf"
+}
 
-	mkdir -p /run/systemd/system/systemd-networkd.service.d
+# fix_systemd_mask_audit masks the systemd-journal-audit socket
+fix_systemd_mask_audit() {
+	ln -s /dev/null /run/systemd/system/systemd-journal-audit.socket
```

Comment:
I believe it would be better if we did `ln -fs`, otherwise our generator will fail when running `systemctl daemon-reload` since the symlink already exists. Or does systemd clear `/run/systemd` before reloading?

Reply:
Changed